otherlibs/systhreads/st_stubs.c

```diff
@@ -545,16 +553,29 @@ static ST_THREAD_FUNCTION caml_thread_start(void * arg)
   value clos;
 #ifdef NATIVE_CODE
   struct longjmp_buffer termination_buf;
-  char tos;
+  void* signal_stack;
+  stack_t signal_stack_t;
+
+  /* Record top of stack and stack size */
+  st_determine_thread_stack_size((void**) &th->top_of_stack, &th->stack_size);
 #endif
 
   /* Associate the thread descriptor with the thread */
   st_tls_set(thread_descriptor_key, (void *) th);
   /* Acquire the global mutex */
   caml_leave_blocking_section();
 #ifdef NATIVE_CODE
-  /* Record top of stack (approximative) */
-  th->top_of_stack = &tos;
+  /* Establish stack to be used for running signal handlers (in particular
```

Comment:
No need to set an alternate stack if `HAS_STACK_OVERFLOW_DETECTION` is not defined.